visivo/parsers/evaluator.py

```diff
@@ -0,0 +1,143 @@
+from typing import Any
+import ast
+import operator as op
+import os
+
+SUPPORTED_NUMPY_FUNCTIONS = ["sum", "all", "mean"]
+
+
+def evaluate_expression(expression: str, project: Any, output_dir: str) -> Any:
+    import numpy
+
+    operators = {
+        ast.Add: op.add,
+        ast.Sub: op.sub,
+        ast.Mult: op.mul,
+        ast.Div: op.truediv,
+        ast.Pow: op.pow,
+        ast.BitXor: op.xor,
+        ast.USub: op.neg,
+    }
+
+    def any_test_failed():
+        # This is a placeholder. Replace with actual implementation.
+        return False
+
+    def build_list_of_nodes(node, previous_nodes=[]):
+        if isinstance(node, ast.Subscript) or isinstance(node, ast.Attribute):
+            previous_nodes.append(node)
+            return build_list_of_nodes(node.value, previous_nodes)
+        elif isinstance(node, ast.Name):
+            previous_nodes.append(node)
+            return previous_nodes
+        else:
+            return previous_nodes
+
+    def get_object_from_node(current_object, index, nodes):
+        node = nodes[index]
+        last_node = index == len(nodes) - 1
+        next_index = index + 1
+        if isinstance(node, ast.Subscript):
+            value = current_object[node.slice.value]
+            if last_node:
+                return value
+            return get_object_from_node(value, next_index, nodes)
+        elif isinstance(node, ast.Attribute):
+            value = getattr(current_object, node.attr)
+            if last_node:
+                return value
+            return get_object_from_node(value, next_index, nodes)
+        elif isinstance(node, ast.Name):
+            return current_object
+        else:
+            raise ValueError(f"Unsupported node type: {node.__class__.__name__}")
+
+    def eval_subscripts(node):
+        nodes = build_list_of_nodes(node)
+        nodes = nodes[::-1]  # Reverse the order of nodes
+        if hasattr(nodes[0], "id") and nodes[0].id == "project":
+            value = get_object_from_node(project, 1, nodes)
+            return value
+        else:
+            raise ValueError(f"Unsupported node type: {nodes[0].id}")
```

Comment:
What's a node? Is that an object in the dag of your project?

Reply:
A node in this context is an node in the Abstract Syntax Tree (AST).  This is how language gets parsed, so I am finding what can be done in our EvalStrings